### PR TITLE
Refactor services to use boosting framework used by UniProtKB

### PR DESCRIPTION
**Pull request by eddturner**
UniProtKB boosting framework (edismax + external config defining boosts) has is more configurable and has given users better search results. The goal of this PR is to ensure all appropriate REST services use the same framework.

### CODE REVIEW

File: common-rest/src/main/java/org/uniprot/api/rest/service/BasicSearchService.java [hunk 1]
```diff
@@ -161,19 +160,15 @@ protected SolrRequest createSolrRequest(SearchRequest request, boolean includeFa
...
-            DefaultSearchHandler defaultSearchHandler,
-            boolean includeFacets) {
+            boolean includeFacets,
+            QueryBoosts queryBoosts) {
 
         SolrRequest.SolrRequestBuilder requestBuilder = SolrRequest.builder();
 
         String requestedQuery = request.getQuery();
 
         boolean hasScore = false;
```
Comment (LeonardoGonzales) [inline common-rest/src/main/java/org/uniprot/api/rest/service/BasicSearchService.java]: I think we do not use hasScore in this method, right? can we remove this line?

File: support-data-rest/src/main/java/org/uniprot/api/disease/DiseaseService.java [hunk 2]
```diff
@@ -1,34 +1,27 @@
...
 import org.springframework.stereotype.Service;
+import org.uniprot.api.common.repository.search.QueryBoosts;
 import org.uniprot.api.rest.service.BasicSearchService;
 import org.uniprot.core.cv.disease.Disease;
-import org.uniprot.store.search.DefaultSearchHandler;
 import org.uniprot.store.search.document.disease.DiseaseDocument;
 import org.uniprot.store.search.field.UniProtSearchFields;
 
 @Service
+@Import(DiseaseQueryBoostsConfig.class)
```
Comment (LeonardoGonzales) [inline support-data-rest/src/main/java/org/uniprot/api/disease/DiseaseService.java]: Do we need to import it?

Comment (eddturner) [reply support-data-rest/src/main/java/org/uniprot/api/disease/DiseaseService.java]: What are your thoughts? For annotation-based configs, then yes, @Import is a normal way of using them.

File: common-rest/src/main/java/org/uniprot/api/rest/service/BasicSearchService.java (hunk 1 above)
Comment (eddturner) [reply common-rest/src/main/java/org/uniprot/api/rest/service/BasicSearchService.java]: it's used a few lines below:

```java
        if (solrSortClause != null) {
            requestBuilder.addSort(solrSortClause.getSort(request.getSort(), hasScore));
        }
```

**Pull request merged**